src/app/qgsdatasourcemanagerdialog.h

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+    qgsdatasourcemanagerdialog.h - datasource manager dialog
+
+    ---------------------
+    begin                : May 19, 2017
+    copyright            : (C) 2017 by Alessandro Pasotti
+    email                : apasotti at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDATASOURCEMANAGERDIALOG_H
+#define QGSDATASOURCEMANAGERDIALOG_H
+
+#include <QList>
+#include <QDialog>
+#include "qgsoptionsdialogbase.h"
+#include "qgsguiutils.h"
+
+class QgsBrowserDockWidget;
+class QgsRasterLayer;
+class QgsMapCanvas;
+
+namespace Ui
+{
+  class QgsDataSourceManagerDialog;
+}
+
+class QgsDataSourceManagerDialog : public QgsOptionsDialogBase
```

Comment:
@manisandro While we're talking about the afs provider, any chance you could get sourcepole to look into removing https://github.com/qgis/QGIS/blob/master/src/providers/arcgisrest/qgsafsfeatureiterator.h#L33 ? It's presence makes the provider very crashy, since iterator lifetime is not tied to the lifetime of a provider and it's essential that iterators can continue to work correctly when the provider is removed.